modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+|===
+
+[discrete]
+== `InventoryItemStatus`
+
+Creates a count and icon for an inventory card with optional link address; used within `InventoryItemBody`
+
+[source,tsx]
+----
+ return (
+   <InventoryItem>
+     <InventoryItemTitle>{title}</InventoryItemTitle>
+     <InventoryItemBody error={loadError}>
+       {loaded && <InventoryItemStatus count={workerNodes.length} icon={<MonitoringIcon />} />}
+     </InventoryItemBody>
+   </InventoryItem>
+ )
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`count` |count for display
+|`icon` |icon for display
+|`linkTo` |(optional) link address
+|===
+
+[discrete]
+== `InventoryItemLoading`
+
+Creates a skeleton container for when an inventory card is loading; used with `InventoryItem` and related components
+
+[source,tsx]
+----
+if (loadError) {
+   title = <Link to={workerNodesLink}>{t('Worker Nodes')}</Link>;
+} else if (!loaded) {
+  title = <><InventoryItemLoading /><Link to={workerNodesLink}>{t('Worker Nodes')}</Link></>;
+}
+return (
+  <InventoryItem>
+    <InventoryItemTitle>{title}</InventoryItemTitle>
+  </InventoryItem>
+)
+----
+
+[discrete]
+== `useFlag`
+
+Hook that returns the given feature flag from FLAGS redux state. It returns the boolean value of the requested feature flag or undefined.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`flag` |The feature flag to return
+|===
+
+[discrete]
+== `YAMLEditor`
+
+A basic lazy loaded YAML editor with hover help and completion.
+
+[source,tsx]
+----
+<React.Suspense fallback={<LoadingBox />}>
+  <YAMLEditor
+    value={code}
+  />
+</React.Suspense>
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`value` |String representing the yaml code to render.
+
+|`options` |Monaco editor options. For more details, please, visit
+https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.IStandaloneEditorConstructionOptions.html.
+
+|`minHeight` |Minimum editor height in valid CSS height values.
+
+|`showShortcuts` |Boolean to show shortcuts on top of the editor.
+
+|`toolbarLinks` |Array of ReactNode rendered on the toolbar links
+section on top of the editor.
+
+|`onChange` |Callback for on code change event.
+
+|`onSave` |Callback called when the command CTRL / CMD + S is triggered.
+
+|`ref` |React reference to `{ editor?: IStandaloneCodeEditor }`. Using
+the 'editor' property, you are able to access to all methods to control
+the editor. For more information, visit
+https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.IStandaloneCodeEditor.html.
+|===
+
+[discrete]
+== `ResourceYAMLEditor`
+
+A lazy loaded YAML editor for Kubernetes resources with hover help and completion. The component use the YAMLEditor and add on top of it more functionality likeresource update handling, alerts, save, cancel and reload buttons, accessibility and more. Unless `onSave` callback is provided, the resource update is automatically handled.It should be
```

Comment:
s/component use/component uses
s/YAMLEditor/YAML editor
s/add on top of it more functionality like/adds functionality, such as
s/resource update handling, alerts, save, cancel and reload buttons, accessibility and more./resource update handling, alerts, save; cancel and reload buttons; and accessibility.
s/ handled.It/handled. It